backend/controllers/SubscriptionsController.ts

```diff
@@ -18,6 +18,41 @@ const SubscriptionsController = {
     } catch (err: any) {
       return res.status(500).json({ error: req.t('other.server_error'), err });
     }
+  },
+  async toggleSubscriptions(req: Request, res: Response) {
+    try {
+      // userId - id of a user whose subscriptions we are interested in
+      // and subscriptionIdm which we want to toggle
+      const { userId, subscriptionId } = req.body;
+
+      if (!userId || !subscriptionId) {
+        return res
+          .status(400)
+          .json({ error: req.t('subsriptions.fields_error') });
+      }
+
+      let userData = await User.findOne({
+        _id: userId
+      });
+
+      if (!userData) {
+        return res.status(400).json({ error: req.t('auth.user_not_exist') });
+      }
+
+      if (userData.subscriptions.includes(subscriptionId)) {
+        userData.subscriptions = userData.subscriptions.filter(
+          s => s !== subscriptionId
+        );
+      } else {
+        userData.subscriptions.push(subscriptionId);
+      }
+
+      await userData.save();
+
+      return res.status(200).json({ subscriptions: userData.subscriptions });
+    } catch (err: any) {
+      return res.status(500).json({ error: req.t('other.server_error'), err });
+    }
```

Comment:
I think userId needs to be saved subscriptionId user model as subscriber
![subscribers](https://user-images.githubusercontent.com/88610270/173055059-be9db6f4-5d59-42ea-bd83-d8e0a3d5a3dc.png)

Reply:
we should discuss about this - i think roman right